### ISSUE TITLE
DuckDB: AL03 on `COLUMNS` expression, `EXCLUDE`, and `REPLACE`

### ISSUE THREAD

**Issue by keraion** [sqlfluff/sqlfluff#5555]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linter returns a failure with rule AL03 for the wildcard expressions with `EXCLUDE` and/or `REPLACE` and `COLUMNS` expressions. Aliases are invalid for these select clause elements.

### Expected Behaviour

The linter should ignore the wildcard and columns expressions for AL03.

### Observed Behaviour

```sh
echo "SELECT COLUMNS(c -> c LIKE '%num%'), 1 AS x FROM numbers;" | sqlfluff lint --dialect duckdb --rules AL03 -
== [stdin] FAIL                                                                                                                                
L:   1 | P:   8 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
All Finished 📜 🎉!

echo "select * exclude (y), 6 as z from tabx;" | sqlfluff lint --dialect duckdb --rules AL03 -
== [stdin] FAIL                                                                                                                                
L:   1 | P:   8 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
All Finished 📜 🎉!

echo "select * replace (3 as x), 6 as z from tabx;" | sqlfluff lint --dialect duckdb --rules AL03 -
== [stdin] FAIL                                                                                                                                
L:   1 | P:   8 | AL03 | Column expression without alias. Use explicit `AS`
                       | clause. [aliasing.expression]
All Finished 📜 🎉!
```

### How to reproduce

```sh
echo "SELECT COLUMNS(c -> c LIKE '%num%'), 1 AS x FROM numbers;" | sqlfluff lint --dialect duckdb --rules AL03 -
echo "select * exclude (y), 6 as z from tabx;" | sqlfluff lint --dialect duckdb --rules AL03 -
echo "select * replace (3 as x), 6 as z from tabx;" | sqlfluff lint --dialect duckdb --rules AL03 -
```

### Dialect

DuckDB

### Version

Python 3.11.7
sqlfluff, version 3.0.0a4

### Configuration

Defaults

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
